### PR TITLE
internal metrics api endpoints

### LINKED ISSUE #681: Performance improvements for self_stats
Split out the more intensive operations (eg I think the check freshness distribution) to separate pages, and separate metrics json endpoints, so it's much lighter to view the basic info. Similarly for self_stats.json (and probably move these to the jsonapi).

### PR DESCRIPTION
**Pull request by jessereynolds**
fixes #681

### CODE REVIEW

**Comment by jessereynolds** [comment]
The changes to changelog and version.rb are actually previously committed to maint/1.x but i hadn't pushed before creating this PR

File: lib/flapjack/gateways/jsonapi/metrics_methods.rb [hunk 1]
```diff
@@ -30,10 +30,16 @@ def event_queue_length
             redis.llen('events')
           end
 
+          def event_queue
+            { 'length' => redis.llen('events') }
+          end
+
           def processed_events
```
Comment (Hobbsee) [inline lib/flapjack/gateways/jsonapi/metrics_methods.rb]: This and the next method seem like they should be put together, unless you're intending to add more parameters here at some point in future.

**Comment by Hobbsee** [comment]
Looks good to me!

File: lib/flapjack/gateways/jsonapi/metrics_methods.rb (hunk 1 above)
Comment (jessereynolds) [inline lib/flapjack/gateways/jsonapi/metrics_methods.rb]: Hmm. I've documented these two endpoints separately as follows:

- [event_queue](http://flapjack.io/docs/1.0/jsonapi/?ruby#event-queue)
- [processed_events](http://flapjack.io/docs/1.0/jsonapi/?ruby#processed-events)

The event_queue endpoint could have other things added in the future, but maybe you're right. Would you combine them like this perhaps? 

## GET /metrics/events
```json
{
  "events": 
  {
    "processed": {
      "all": 153,
      "failure": 104,
      "ok": 12,
      "invalid": 29,
      "action": 8
    }
    "queue": {
      "length": 0
    }
  }
}
```

I'm not sure it really give us anything.

**Comment by jessereynolds** [comment]
I'm reworking this to merge the two events endpoints as described above.

**Comment by jessereynolds** [comment]
Don't merge yet - want to add functions to diner and get pacts in place

File: lib/flapjack/gateways/jsonapi/metrics_methods.rb (hunk 1 above)
Comment (Hobbsee) [inline lib/flapjack/gateways/jsonapi/metrics_methods.rb]: As you have it documented in those links makes sense - it just looks like you have the same command from event_queue_length, just wrapped in a hash, making it event_queue.

**Comment by jessereynolds** [comment]
I haven't done diner pact tests for all of these endpoints yet but this is good to merge as is I think.

**Pull request merged**